core/src/main/scala/kafka/zk/KafkaZkClient.scala

```diff
@@ -49,6 +49,32 @@ import scala.collection.{Seq, mutable}
 class KafkaZkClient(zooKeeperClient: ZooKeeperClient, isSecure: Boolean) extends Logging {
   import KafkaZkClient._
 
+  def createSequentialPersistentPath(path: String, data: String = ""): String = {
+    val createRequest = CreateRequest(path, data.getBytes("UTF-8"), acls(path), CreateMode.PERSISTENT_SEQUENTIAL)
+    val createResponse = retryRequestUntilConnected(createRequest)
+    if (createResponse.resultCode != Code.OK) {
+      createResponse.resultException.foreach(e => throw e)
+    }
+    createResponse.path
+  }
+
+  def leaderAndIsrZkData(leaderAndIsr: LeaderAndIsr, controllerEpoch: Int): String = {
```

Comment:
This method doesn't access ZK. So, it should be in ZkData.scala.

Reply:
I looked at ZkData.scala where there are bunch of objects.
Not sure whether this method fits there.

Reply:
We could just add a new LeaderAndIsr object and put the method there.